src/server/express/http_rest_handler.ts

```diff
@@ -0,0 +1,477 @@
+import express, {
+  Request,
+  Response,
+  Router,
+  RequestHandler,
+  ErrorRequestHandler,
+  NextFunction,
+} from 'express';
+import { A2ARequestHandler } from '../request_handler/a2a_request_handler.js';
+import { A2AError } from '../error.js';
+import {
+  Message,
+  Task,
+  TaskStatusUpdateEvent,
+  TaskArtifactUpdateEvent,
+  TaskQueryParams,
+  TaskIdParams,
+  MessageSendParams,
+} from '../../types.js';
+import { RestMessageSendParams, RestTaskPushNotificationConfig } from './rest_types.js';
+import {
+  fromRestMessageSendParams,
+  fromRestTaskPushNotificationConfig,
+  toRestMessage,
+  toRestTask,
+  toRestStreamEvent,
+  toRestTaskPushNotificationConfig,
+  HTTP_STATUS,
+  mapErrorToStatus,
+  parseHistoryLength,
+  extractAction,
+  SSE_HEADERS,
+  formatSSEEvent,
+  formatSSEErrorEvent,
+  ACTION,
+} from '../transports/http_rest_transport_handler.js';
+
+/**
+ * Options for configuring the HTTP REST handler.
+ */
+export interface HttpRestHandlerOptions {
+  /** The A2A request handler implementation that processes requests */
+  requestHandler: A2ARequestHandler;
+}
+
+/**
+ * Express error handler middleware for REST API JSON parse errors.
+ * Catches SyntaxError from express.json() and converts to A2A parse error format.
+ *
+ * @param err - Error thrown by express.json() middleware
+ * @param _req - Express request (unused)
+ * @param res - Express response
+ * @param next - Next middleware function
+ */
+export const restErrorHandler: ErrorRequestHandler = (
+  err: Error,
+  _req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (err instanceof SyntaxError && 'body' in err) {
+    const a2aError = A2AError.parseError('Invalid JSON payload.');
+    return res.status(400).json({
+      error: a2aError.toJSONRPCError(),
+    });
+  }
+  next(err);
+};
+
+/**
+ * Regular expression patterns for matching REST API routes with actions.
+ */
+const ROUTE_PATTERN = {
+  MESSAGE_ACTION: /^\/v1\/message:(send|stream)$/i,
+  TASK_ACTION: /^\/v1\/tasks\/([^/:]+):([a-z]+)$/i,
+} as const;
```

Comment:
Let's consider simplifying it so that single `router.get`/`route.post` handles one method to make the logic more explicit and avoid advanced regex matching. Code duplication between similar "groups" of actions can be extracted into helper methods.